### PR TITLE
Added downloading proper file extensions

### PR DESCRIPTION
**Pull request by nate-moo**
For example, if a video stream source is m3u8 intead of downloading the file as x.m3u8 and renaming it to {anime_title}/{anime_title}_{ep_no}.mp4, we download it as x.m3u8 and dont rename the extension so the extension will stay but the name will change to the configured value: {anime_title}/{anime_title}_{ep_no}.m3u8. (except for nyaa which I have set to always be mkv bc that is almost always the file ext used)

### CODE REVIEW

**Review by IguanasInPyjamas — APPROVED** [review]
This is fine, someone should do some testing first.

**Comment by nate-moo** [comment]
> This is fine, someone should do some testing first.

tested, also found out alot of sites dont work lmao

gogoanime: works,

kisscartoon: got EOF from server?,

twist: works, 

animeflv: No supported Host on the 3 tested maybe broken?,

Itsaturday: nonetype error,

animefreak: Network error? maybe bc im using VPN 
(Edit: nope)

animeout: works, 

animixplay: works, 

animesimple: has other issues, 

ryuanime: has other issues, 

animerush: has issues with node sandboxing,

animeflix: resource not found?,

kickass: works,

Dreamanime: times out,

erai-raws: 403 forbidden?,

watchmovie: works,

animekisa: node sandbox issue,

animedaisuki: vidlink domain name not found,

animevibe: cloudflare?,

yify: fuck yify... it timed out,

animefree: http 400 bad request?

voiranime: node sandbox issue,

vidstream: works

**Comment by Blatzar** [comment]
I'll do some tweaking to solve https://github.com/vn-ki/anime-downloader/issues/428 too.

**Comment by nate-moo** [comment]
> I'll do some tweaking to solve #428 too.

you could essentially just not rename it when downloading from nyaa or a downloader specific site like animeout

**Comment by nate-moo** [comment]
waka waka waka waka waka waka waka waka I'm pacman

File: anime_downloader/util.py [hunk 1]
```diff
@@ -268,25 +268,31 @@ def format_command(cmd, episode, file_format, speed_limit, path):
 
     cmd_dict = {
         '{aria2}': 'aria2c {stream_url} -x 12 -s 12 -j 12 -k 10M -o '
-                   '{file_format}.mp4 --continue=true --dir={download_dir}'
+                   '{file_format}{file_ext} --continue=true --dir={download_dir}'
                    ' --stream-piece-selector=inorder --min-split-size=5M --referer={referer} --check-certificate=false --user-agent={useragent} --max-overall-download-limit={speed_limit}',
         '{idm}'  : 'idman.exe /n /d {stream_url} /p {download_dir} /f {file_format}.mp4'
     }
 
+    file_name, file_ext = os.path.splitext(episode.source().stream_url)
```
Comment (Blatzar) [inline anime_downloader/util.py:276]: Get the final redirect url with requests.get(episode.source().stream_url, stream=True)

File: anime_downloader/util.py [hunk 2]
```diff
@@ -268,25 +268,31 @@ def format_command(cmd, episode, file_format, speed_limit, path):
...
+    file_ext = file_ext.split('?', 1)[0]
```
Comment (Blatzar) [inline anime_downloader/util.py:277]: Use urllib instead

File: anime_downloader/util.py [hunk 3]
```diff
@@ -268,25 +268,31 @@ def format_command(cmd, episode, file_format, speed_limit, path):
...
+    if file_ext == ".php" or file_ext == "" or file_ext == "html":
```
Comment (Blatzar) [inline anime_downloader/util.py:278]: Use a dict/list instead of tons of or statements.

File: anime_downloader/util.py [hunk 4]
```diff
@@ -268,25 +268,31 @@ def format_command(cmd, episode, file_format, speed_limit, path):
...
+        file_ext = ".mp4"
+
     # Allows for passing the user agent with self.headers in the site.
     # Some sites block downloads using a different user agent.
     if episode.headers.get('user-agent'):
         useragent = episode.headers['user-agent']
     else:
         useragent = 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36'
-
+
```
Comment (Blatzar) [inline anime_downloader/util.py:287]: No extra spaces

**Review by Blatzar — CHANGES_REQUESTED** [review]

**Comment by IguanasInPyjamas** [comment]
Closing in favor of #519

**Comment by nate-moo** [comment]
ah yes in favor of something that hasn't gone anywhere in a while 
*I also haven't done anything with this so it's partially my fault*

**Pull request closed without merging**